eras/shelley/impl/src/Cardano/Ledger/Shelley/PParams.hs

```diff
@@ -212,27 +212,8 @@ instance EraPParams ShelleyEra where
 
   pparams = shelleyPParams
 
-instance Era era => EncCBOR (ShelleyPParams Identity era) where
-  encCBOR
-    ShelleyPParams {..} =
-      encodeListLen 17
-        <> encCBOR sppMinFeeA
-        <> encCBOR sppMinFeeB
-        <> encCBOR sppMaxBBSize
-        <> encCBOR sppMaxTxSize
-        <> encCBOR sppMaxBHSize
-        <> encCBOR sppKeyDeposit
-        <> encCBOR sppPoolDeposit
-        <> encCBOR sppEMax
-        <> encCBOR sppNOpt
-        <> encCBOR sppA0
-        <> encCBOR sppRho
-        <> encCBOR sppTau
-        <> encCBOR sppD
-        <> encCBOR sppExtraEntropy
-        <> encCBOR sppProtocolVersion
-        <> encCBOR sppMinUTxOValue
-        <> encCBOR sppMinPoolCost
+instance EraPParams era => EncCBOR (ShelleyPParams Identity era) -- should this be: encCBOR p = encCBOR (PParams @era p) instead?
```

Comment:
I'm at a loss with this, as I wrote in the comment. 
Is it safer to explicitly define the implementation as such:
`encCBOR p = encCBOR (PParams @era p)` 
and add this constraint? 
`PParamsHKD Identity era ~ ShelleyPParams Identity era`

Reply:
You do not need this instance anymore, because what we need is `EncCBOR (PParams era)`, which you provided in `cardano-ledger-core`. The part that you are missing here is that it is still required here: https://github.com/IntersectMBO/cardano-ledger/blob/d90eb4df4651970972d860e95f1a3697a3de8977/libs/cardano-ledger-core/src/Cardano/Ledger/Core/PParams.hs#L236 which you can safely remove as well.

Same comment applies to all serialization instances for PParamsHKD type family.